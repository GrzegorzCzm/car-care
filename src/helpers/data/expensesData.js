const expenses = [
  {
    id: 0, date: '2018-04-19', item: 'tires', cost: 35.45, description: 'some details here', address: 'dummy address', position: { lat: 51.109372, lng: 16.958476 },
  },
  {
    id: 1, date: '2018-08-13', item: 'some long name here', cost: 85, description: 'some more details here', address: 'dummy second address', position: { lat: 51.109372, lng: 16.958476 },
  },
  {
    id: 2, date: '2019-07-10', item: 'wheels', cost: 1450.65, description: 'some more details here', address: 'dummy third address', position: { lat: 51.109372, lng: 16.958476 },
  },
];


export default expenses;
