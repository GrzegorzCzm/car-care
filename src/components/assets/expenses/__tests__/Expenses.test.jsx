import React from 'react';
import { Provider } from 'react-redux';
import { shallow, mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';

import expensesData from '../../../../helpers/data/expensesData';
import ConnectedExpenses, { Expenses } from '../Expenses';


describe('Expenses', () => {
  const props = {
    expenses: expensesData,
    classes: {
      listRoot: 'expenses-list',
      paramLine: 'expenses-paramLine',
    },
  };

  it('should renders correctly with compact list', () => {
    const wrapper = shallow(<Expenses {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should renders correctly with full list', () => {
    const wrapper = shallow(<Expenses {...props} />);
    const switchButtonControlProps = wrapper.find('.switch-button').props('control').control.props;
    switchButtonControlProps.onChange();

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should contains classes prop and store with expenses list', () => {
    const mockStore = configureStore([]);
    const store = mockStore({
      expenses: expensesData,
    });

    const expectedClasses = { listRoot: 'Expenses-listRoot-1', paramLine: 'Expenses-paramLine-2' };

    const wrapper = mount(
      <Provider store={store}>
        <ConnectedExpenses />
      </Provider>,
    );

    expect(wrapper.find('Expenses').prop('classes')).toEqual(expectedClasses);
    expect(wrapper.find('Provider').prop('store').getState().expenses.length).toBe(3);
  });

  it('should change isFullView when setFullView clicked', () => {
    const wrapper = shallow(<Expenses {...props} />);
    const switchButtonControlProps = wrapper.find('.switch-button').props('control').control.props;

    expect(switchButtonControlProps.checked).toBe(false);
    switchButtonControlProps.onChange();
    expect(switchButtonControlProps.checked).toBe(false);
  });
});
