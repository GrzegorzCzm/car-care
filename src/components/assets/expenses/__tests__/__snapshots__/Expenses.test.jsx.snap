// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expenses should renders correctly with compact list 1`] = `
<Styled(MuiBox)>
  <Styled(MuiBox)
    className="view-switch"
  >
    <WithStyles(ForwardRef(FormControlLabel))
      className="switch-button"
      control={
        <ForwardRef(WithStyles)
          checked={false}
          color="primary"
          onChange={[Function]}
          value="Switch"
        />
      }
      label="Full view"
    />
  </Styled(MuiBox)>
  <Styled(MuiBox)
    className="expenses-list"
  >
    <WithStyles(ForwardRef(List))
      className="expenses-list"
    >
      <WithStyles(SingleExpenseCompact)
        expense={
          Object {
            "address": "dummy address",
            "cost": 35.45,
            "date": "2018-08-19",
            "description": "some details here",
            "id": 0,
            "item": "tires",
            "position": Object {
              "lat": 51.109372,
              "lng": 16.958476,
            },
          }
        }
        key="0"
      />
      <WithStyles(SingleExpenseCompact)
        expense={
          Object {
            "address": "dummy second address",
            "cost": 85,
            "date": "2018-08-13",
            "description": "some more details here",
            "id": 1,
            "item": "some long name here",
            "position": Object {
              "lat": 51.109372,
              "lng": 16.958476,
            },
          }
        }
        key="1"
      />
      <WithStyles(SingleExpenseCompact)
        expense={
          Object {
            "address": "dummy third address",
            "cost": 1450.65,
            "date": "2018-07-10",
            "description": "some more details here",
            "id": 2,
            "item": "wheels",
            "position": Object {
              "lat": 51.109372,
              "lng": 16.958476,
            },
          }
        }
        key="2"
      />
    </WithStyles(ForwardRef(List))>
  </Styled(MuiBox)>
</Styled(MuiBox)>
`;

exports[`Expenses should renders correctly with full list 1`] = `
<Styled(MuiBox)>
  <Styled(MuiBox)
    className="view-switch"
  >
    <WithStyles(ForwardRef(FormControlLabel))
      className="switch-button"
      control={
        <ForwardRef(WithStyles)
          checked={true}
          color="primary"
          onChange={[Function]}
          value="Switch"
        />
      }
      label="Full view"
    />
  </Styled(MuiBox)>
  <Styled(MuiBox)
    className="expenses-list"
  >
    <WithStyles(SingleExpense)
      expense={
        Object {
          "address": "dummy address",
          "cost": 35.45,
          "date": "2018-08-19",
          "description": "some details here",
          "id": 0,
          "item": "tires",
          "position": Object {
            "lat": 51.109372,
            "lng": 16.958476,
          },
        }
      }
      key="0"
    />
    <WithStyles(SingleExpense)
      expense={
        Object {
          "address": "dummy second address",
          "cost": 85,
          "date": "2018-08-13",
          "description": "some more details here",
          "id": 1,
          "item": "some long name here",
          "position": Object {
            "lat": 51.109372,
            "lng": 16.958476,
          },
        }
      }
      key="1"
    />
    <WithStyles(SingleExpense)
      expense={
        Object {
          "address": "dummy third address",
          "cost": 1450.65,
          "date": "2018-07-10",
          "description": "some more details here",
          "id": 2,
          "item": "wheels",
          "position": Object {
            "lat": 51.109372,
            "lng": 16.958476,
          },
        }
      }
      key="2"
    />
  </Styled(MuiBox)>
</Styled(MuiBox)>
`;
